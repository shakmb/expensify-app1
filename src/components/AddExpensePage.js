import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { addExpense } from "../actions/expenses";

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    // props.dispatch(addExpense(expense));
    this.props.addExpense(expense);
    this.props.history.push("/");
  };
  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <ExpenseForm onSubmit={this.onSubmit} />
      </div>
    );
  }
}

// converted from function component to class during the tutorial to avoid inline function handlers
// const AddExpensePage = (props) => (
//   <div>
//     {/* This is from my add expense component */}
//     <h1>Add Expense</h1>
//     <ExpenseForm
//       onSubmit={(expense) => {
//         // props.dispatch(addExpense(expense));
//         props.onSubmit(expense);
//         props.history.push("/");
//       }}
//     />
//   </div>
// );

const mapDispatchToProps = (dispatch) => ({
  addExpense: (expense) => dispatch(addExpense(expense)),
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
