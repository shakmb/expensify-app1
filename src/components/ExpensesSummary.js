import React from "react";
import { connect } from "react-redux";
import selectExpenses from "../selectors/expenses";
import selectExpensesTotal from "../selectors/expenses-total";
import numeral from "numeral";

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  return (
    <div>
      <h1>
        Viewing {expenseCount} {expenseCount !== 1 ? "expenses" : "expense"}{" "}
        totalling {numeral(expensesTotal / 100).format("$0,0.00")}
      </h1>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
