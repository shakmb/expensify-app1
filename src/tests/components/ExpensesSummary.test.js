import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";
import expenses from "../fixtures/expenses";

test("should return the correct count and total for 1 expense", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={1} expensesTotal={12940} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should return the correct count and total for 2 expenses", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={2} expensesTotal={2000} />
  );
  expect(wrapper).toMatchSnapshot();
});
