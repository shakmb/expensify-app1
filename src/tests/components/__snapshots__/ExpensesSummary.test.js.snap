// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return the correct count and total for 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     
    totalling 
    $129.40
  </h1>
</div>
`;

exports[`should return the correct count and total for 2 expenses 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     
    totalling 
    $20.00
  </h1>
</div>
`;
